spec/page_spec.rb

```diff
@@ -124,5 +124,32 @@
       EOS
       (expect pdf.positions[0]).to eql [72.0, 757.926]
     end
+
+    it 'should change the margin if the layout changes' do
+      pdf = to_pdf <<~'EOS', analyze: :text
+      portrait-1
+
+      <<<
+
+      portrait-2
+
+      [.landscape]
+      <<<
+
+      landscape-1
+
+      <<<
+
+      landscape-2
+      EOS
+      (expect pdf.positions[0]).to eql [48.24, 793.926]
```

Comment:
This is better done using `pdf.find_text`, then look at the x/y position of the results.